MadDonkeySoftware/corp-hq
# CHQ-49 Added authentication API

### Story
https://maddonkeysoftware.myjetbrains.com/youtrack/issue/CHQ-49

#### Depends On (Optional)
https://github.com/MadDonkeySoftware/corp-hq/pull/20

### Summary
* Added AuthenticationController and supporting classes
* Modified `User` password and salt fields from `string` to `byte[]`
  * This is because the string wasn't being translated back to a byte array properly for some reason.


### Testing
* Optionally: drop the users table
* Register a user that you know the user name and password for
* Make a good request to verify you get a HTTP 200
```
curl -X POST http://localhost:5000/api/v1/authentication \
  -H 'Content-Type: application/json' \
  -d '{
	"username": "test",
	"password": "testtest"
}'
```
* Make a good request to verify you get a HTTP 401
```
curl -X POST http://localhost:5000/api/v1/authentication \
  -H 'Content-Type: application/json' \
  -d '{
	"username": "tes",
	"password": "testtest"
}'
```
```
curl -X POST http://localhost:5000/api/v1/authentication \
  -H 'Content-Type: application/json' \
  -d '{
	"username": "test",
	"password": "testtest123"
}'
```
